Ginger/Unit Tests/NonUITests/AutoPilot/APILearningTests.cs

```diff
@@ -0,0 +1,144 @@
+            TemplateFile TempleteFile = new TemplateFile() { FilePath = createPaymentProfileFileName };
+            createPaymentProfileModels[0].OptionalValuesTemplates.Add(TempleteFile);
+            Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict = new Dictionary<Tuple<string, string>, List<string>>();
+            ImportOptionalValuesForParameters ImportOptionalValues = new ImportOptionalValuesForParameters();
+            ImportOptionalValues.ShowMessage = false;
+            ImportOptionalValues.GetAllOptionalValuesFromExamplesFiles(createPaymentProfileModels[0], OptionalValuesPerParameterDict);
+            ImportOptionalValues.PopulateOptionalValuesForAPIParameters(createPaymentProfileModels[0], OptionalValuesPerParameterDict);
+
+            //Assert
+            Assert.AreEqual(createPaymentProfileModels.Count, 1, "APIModels count");
+            Assert.AreEqual(createPaymentProfileModels[0].AppModelParameters.Count, 1, "AppModelParameters count");
+            Assert.AreEqual(createPaymentProfileModels[0].AppModelParameters[0].OptionalValuesList.Count, 1, "AppModelParameters count");
+        }
+
+        [TestMethod]
+        public void JSONTemplateCreateBillingArrangement()
+        {
+            //Arrange                        
+            JSONTemplateParser jSONTemplateParser = new JSONTemplateParser();
+            string createBillingArrangementFileName = TestResources.GetTestResourcesFile(@"AutoPilot\JSONTemplates\login.json");
+
+            //Act   
+            ObservableList<ApplicationAPIModel> createPaymentProfileModels = new ObservableList<ApplicationAPIModel>();
+            jSONTemplateParser.ParseDocument(createBillingArrangementFileName, createPaymentProfileModels);
+            TemplateFile TempleteFile = new TemplateFile() { FilePath = createBillingArrangementFileName };
+            createPaymentProfileModels[0].OptionalValuesTemplates.Add(TempleteFile);
+            Dictionary<Tuple<string, string>, List<string>> OptionalValuesPerParameterDict = new Dictionary<Tuple<string, string>, List<string>>();
+            ImportOptionalValuesForParameters ImportOptionalValues = new ImportOptionalValuesForParameters();
+            ImportOptionalValues.ShowMessage = false;
+            ImportOptionalValues.GetAllOptionalValuesFromExamplesFiles(createPaymentProfileModels[0], OptionalValuesPerParameterDict);
+            ImportOptionalValues.PopulateOptionalValuesForAPIParameters(createPaymentProfileModels[0], OptionalValuesPerParameterDict);
+
+            //Assert
+            Assert.AreEqual(createPaymentProfileModels.Count, 1, "APIModels count");
+            Assert.AreEqual(createPaymentProfileModels[0].AppModelParameters.Count, 2, "AppModelParameters count");
+            Assert.AreEqual(createPaymentProfileModels[0].AppModelParameters[0].OptionalValuesList.Count, 1, "AppModelParameters count");
+        }
```

Comment:
Don't check on count, please also check actual Values value.
same is for all tests in this class